packages/langium/src/services.ts

```diff
@@ -152,6 +153,7 @@ export type LangiumDefaultSharedServices = {
         CodeLensProvider?: CodeLensProvider
         DocumentLinkProvider?: DocumentLinkProvider
         ExecuteCommandHandler?: ExecuteCommandHandler
+        InlayHintProvider?: InlayHintProvider
```

Comment:
Why are this and `CodeLensProvider`, `DocumentLinkProvider` shared service and not language-specific?

Reply:
All of these language services offer a way to "resolve" their respective results. For example, you can return an inlay hint just with it's text and position, and calculate all the other information once requested (for example the tooltip property if the user hovers over it). It's the same for the code lens and document link provider

For these special resolve requests, we don't get the URI of the original document, only the requested code lens, inlay hint or document link. That's the reason why these services are shared.

In theory we could remove the option to resolve these items, that would allow us to move them to the language specific services.

Reply:
> In theory we could remove the option to resolve these items, that would allow us to move them to the language specific services.

Reply:
I've created https://github.com/langium/langium/issues/925.